src/promptflow/promptflow/executor/_service/utils/process_utils.py

```diff
@@ -0,0 +1,107 @@
+# ---------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# ---------------------------------------------------------
+
+import asyncio
+import contextlib
+import json
+import multiprocessing
+import os
+from datetime import timedelta
+from typing import Callable
+
+from promptflow._core._errors import UnexpectedError
+from promptflow._core.operation_context import OperationContext
+from promptflow._utils.exception_utils import ExceptionPresenter, JsonSerializedPromptflowException
+from promptflow._utils.logger_utils import service_logger
+from promptflow.exceptions import ErrorTarget
+from promptflow.executor._service._errors import ExecutionTimeoutError
+from promptflow.executor._service.utils.service_utils import get_log_context
+
+LONG_WAIT_TIMEOUT = timedelta(days=1).total_seconds()
+SHORT_WAIT_TIMEOUT = 10
+
+
+async def invoke_function_in_process(
+    request, context_dict: dict, target_function: Callable, wait_timeout: int = LONG_WAIT_TIMEOUT
+):
+    with multiprocessing.Manager() as manager:
+        return_dict = manager.dict()
+        error_dict = manager.dict()
+
+        p = multiprocessing.Process(
+            target=execute_function,
+            args=(target_function, request, return_dict, error_dict, context_dict),
+        )
+        p.start()
+        service_logger.info(f"[{os.getpid()}--{p.pid}] Start process to execute the request.")
+
+        # Wait for the process to finish or timeout asynchronously
+        await asyncio.get_running_loop().run_in_executor(None, p.join, wait_timeout)
+
+        # Terminate the process if it is still alive after timeout
+        if p.is_alive():
+            service_logger.error(f"[{p.pid}] Stop process for exceeding {wait_timeout} seconds.")
+            p.terminate()
+            p.join()
+            raise ExecutionTimeoutError(wait_timeout)
+
+        # Raise exception if the process exit code is not 0
+        if p.exitcode and p.exitcode > 0:
+            exception = error_dict.get("error", None)
+            if exception is None:
+                raise UnexpectedError(
+                    message="Unexpected error occurred while executing the request",
+                    target=ErrorTarget.EXECUTOR,
+                )
+            # JsonSerializedPromptflowException will be raised here
+            # no need to change to PromptflowException since it will be handled in app.exception_handler
+            raise exception
+
+        service_logger.info(f"[{p.pid}] Process finished.")
+        return return_dict.get("result", {})
+
+
+def execute_function(
```

Comment:
```suggestion
def _execute_function(
```

Reply:
Updated in another PR.
https://github.com/microsoft/promptflow/pull/2065/files